### ISSUE TITLE
More robust `unreachable_code_linter`

### ISSUE THREAD

**Issue by MEO265** [r-lib/lintr#2127]
As soon as there is a comment after the `return` or `stop`, it is correctly not linted, but neither are the following lines.

Will produce lints:
```r
foo  <- function() {
  return(3)
  5 + 2
}
```

False negative: 
```r
foo  <- function() {
  return(3)  # A comment
  5 + 2
}
```
The same applies to something that @MichaelChirico  has already marked as a todo in the code, but I haven't found an issue about it:
https://github.com/r-lib/lintr/blob/c8f00746501e041d2516da58ccca3e054cf305f1/tests/testthat/test-unreachable_code_linter.R#L181
However, not only will the following expression in the same line not be marked, but also those in subsequent lines will not be marked.

False negative: 
```r
foo  <- function() {
  return(3); x + 1
  5 + 2
}
```

**Comment by MEO265** [comment]
I think I've already found a solution and will probably post the PR when I've finished #2123
